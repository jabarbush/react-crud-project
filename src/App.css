@import url('https://fonts.googleapis.com/css?family=Figtree');
@import url('https://fonts.googleapis.com/css?family=Dancing+Script');
@import url('https://fonts.googleapis.com/css?family=Mr+Dafoe');
@import url('https://fonts.googleapis.com/css?family=Great+Vibes');

/* TODO: separate these out to their own files depending on where they're used??? */
.app {
  height: 100vh;
  padding: 2.5rem 1.5rem 1.5rem 1.5rem;
  align-items: flex-start;
  gap: 1.5rem;
  flex: 1 0 0;
  align-self: stretch;
  background-color:#2A2E38;
  font-family: 'Figtree';
  color: #D1D4DC;
}

.page-title {
  display: flex;
  width: 64rem;
  flex-direction: column;
  font-size: 1.125rem;
}

.add-user-btn {
  padding: 0.875rem 1.25rem;
  align-items: flex-start;
  gap: 0.625rem;
  border-radius: 6px;
  border: 1px solid var(--gray-1, #333);
  background: var(--ritten-blue, #1890FF);
  box-shadow: 0px 1px 8px 0px rgba(82, 71, 67, 0.50);
}

.add-user-btn:hover, .signature-view-edit-btn:hover, .add-user-submit:hover,
.cancel:hover, .cancel-signature:hover, .add-signature-submit:hover, .sort-button:hover {
  opacity: 70%;
}

.toolbar {
  margin-bottom: 1rem;
}

.sort-container {
  position: relative;
  display: inline-block;
  
}

.sort-options, .filter-options {
  width: 15rem;
  flex-direction: column;
  align-items: flex-start;
  border-radius: 4px;
  border: 1px solid #2A2E38;
  background: #121623;
  position: absolute;
  z-index: 99;
  right: 10rem;
}

.filter-options {
  width: 17rem;
  left: 19rem;
}

.sort-option, .filter-option {
  padding: 0.62rem;
  font-size: 1rem;
  line-height: 100%;
}

.sort-option:hover, .filter-option:hover, .icon-big-x {
  color: #1890FF;
}

.filter-options {
  right: -5rem;
}

.sort-button {
  padding: 0.5rem 0rem;
  margin: 0 0.5rem 0 0.5rem;
  background-color: transparent;
  border: transparent;
  color: #D1D4DC;
}

.toolbar-title {
 display: inline;
 margin-left: 1rem;
}

.list-container {
  display: block;
  flex-wrap: wrap;
  width: 100%;
  transition: width 0.2s ease;
}

.list-container.active {
  width: 50%;
  transition: width 0.2s ease;
}

.card {
  display: flex;
  justify-content: space-between;
  padding: 1.25rem 1.5rem;
  margin-bottom: 0.75rem;
  margin-right: 0.75rem;
  align-items: center;
  flex: 1 0 0;
  border-radius: 8px;
  border: 1px solid #363A45;
  background: #121623;
  box-shadow: 0px 1px 8px 0px rgba(82, 71, 67, 0.50);
}

.card.active {
  border-radius: 8px;
  border: 1px solid #1890FF;
  background: #121623;
  box-shadow: 0 0 10px 5px rgba(82, 71, 67, 0.50);
  outline: 2px solid #1890FF;
  outline-offset: -2px;
}

.user-name, .toolbar-title, .sort-button, .delete-user-title, .signature-form-title, .edit-user-info-title, .signature-status-title, .signature-status-info, .page-title {
  font-size: 1rem;
  font-weight: 500;
  line-height: 100%;
}

h3 {
  margin-top: 0;
}

.dob-card {
  margin-top: 0.5rem;
  font-size: 0.6875rem;
  line-height: 100%;
}

.add-user-popup {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
  border-radius: 6px;
  background: #121623;
  width: 24.874rem;
  height: 26rem;
}

/* This is to give the background a grayish tint if a pop-up is active*/
.popup-overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 100;
}

.delete-user-overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 100;
}

.delete-user-popup {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
  border-radius: 6px;
  background: #121623;
  width: 17rem;
  height: 10.375rem;
  padding: 1.5rem;
}

.delete-message {
  font-size: 1rem;
  line-height: 100%;
  margin-bottom: 1.8rem;
}

form {
  display: inline-flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  border-radius: 6px;
  background: #121623;
  margin: 1.5rem;
}

label {
  font-size: 0.6875rem;
  line-height: 100%;
  display: flex;
  flex-direction: column;
  flex: 1 0 0;
  padding-bottom: 0.5rem;
}

input, .dropdown {
  width: 100%;
  display: flex;
  padding: 0.6875rem;
  align-items: center;
  flex: 1 0 0;
  color: #D1D4DC;
  border-radius: 4px;
  border: 1px solid #50535E;
  background: #0A0C13;
  margin-bottom: 1rem;
  box-sizing: border-box;
  padding-right: 0;
  margin-right: 0;
}

.new-user-title {
  font-size: 1.125rem;
  font-weight: 700;
  line-height: 100%;
  margin-top: 0;
}

.add-user-submit,
.cancel{
  display: flex;
  align-items: center;
  gap: 0.625rem;
  color: #0A0C13;
  font-size: 0.875rem;
  line-height: 100%;
  padding: 0.875rem 1.25rem;
  border-radius: 6px;
  justify-content: flex-end;
  bottom:1.5rem
}

.add-user-submit, .add-signature-submit{
  border: 1px solid var(--ritten-blue, #1890FF);
  background: var(--ritten-blue, #1890FF);
  box-shadow: 0px 1px 8px 0px rgba(213, 213, 213, 0.50);
}

.add-user-submit {
  position: absolute;
  right: 1.5rem;
}

.cancel, .cancel-signature, .sort-button {
  color: var(--ritten-blue, #1890FF);
  background-color: transparent;
  border: transparent;
}

.cancel {
  position: absolute;
  right: 9rem;
}

.cancel-signature, .add-signature-submit {
  float: right;
  text-align: right;
  display: flex;
  align-items: center;
  gap: 0.625rem;
  font-size: 0.875rem;
  line-height: 100%;
  padding: 0.875rem 1.25rem;
  border-radius: 6px;
  justify-content: flex-end;
  margin: 1rem 0 0 1rem;
}

.add-signature-submit {
  border: 1px solid var(--ritten-blue, #1890FF);
  background: var(--ritten-blue, #1890FF);
  box-shadow: 0px 1px 8px 0px rgba(213, 213, 213, 0.50);
}

.signature-status {
  display: flex;
  padding: 0.25rem 0.5rem;
  align-items: center;
  gap: 0.625rem;
  border-radius: 4px;
  margin-left: auto;
  margin-right: 1rem;
}

.green {
  border: 2px solid #26A69A;
  background: #1B6261;
}

.red {
  border: 2px solid #EF5350;
  background: rgba(239, 83, 80, 0.57);
}

.signature-status-section {
  padding: 1.5rem;
}

.signature-status-title {
  margin-bottom: 0.62rem;
  margin-top: 0;
}

.signature-status-info {
  font-size: 0.875rem;
}

.green-info {
  color: #26A69A;
}

.red-info {
  color: var(--red, #EB5757);
}

.signature-view-edit-btn {
  display: flex;
  margin-top: -2.5rem;
  padding: 0.875rem 1.25rem;
  align-items: flex-end;
  justify-content: flex-end;
  margin-left: auto;
  gap: 0.625rem;
  border-radius: 6px;
  border: 1px solid var(--gray-1, #333);
  background: var(--ritten-blue, #1890FF);
  box-shadow: 0px 1px 8px 0px rgba(82, 71, 67, 0.50);
}

/* TODO: fix this and card list from colliding when zooming in*/
.info-tab {
  position: absolute;
  top: 0;
  right: 0;
  width: 48%;
  background: #121623;
  border-radius: 4px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  margin: 8.62rem 1.5rem 2.5rem 0rem;
  border-radius: 8px;
  border: 1px solid #363A45;
  opacity: 0;
  animation: slideIn 0.2s ease-in-out forwards;
  flex-shrink: 0;
  height: 80%;
}

@keyframes slideIn {
  from {
    opacity: 1;
    transform: translateX(100%);
  }
  to {
    opacity: 1;
    transform: translateX(0);
  }
}

hr {
  width: 100%;
  height: 0.0625rem;
  background: #787878;
  margin: 0;
}

.user-info {
  font-size: 0.875rem;
  font-weight: 700;
  line-height: 100%;
  display: inline-flex;
  margin-top: 1rem;
  padding: 0 1rem 0 0;
  margin-bottom: 0;
}

.user-info-value {
  font-weight: normal;
}

.data {
  align-self: flex-start;
  padding: 1.5rem;
  margin-bottom: 0;
}

.icons {
  position: absolute;
  top: 0;
  right: 0;
  display: flex;
  align-items: center;
  padding: 1.5rem 1.5rem 0rem 0rem;
}

.icon {
  width: 1.25rem;
  padding-left: 1rem;
  height: 1.25rem;
}

.icon-big-x {
  width: 2rem;
  height: 2rem;
  padding-left: 0.5rem;
}

.icon:hover {
  color: #1890ff;
}

.panel-icon {
  color: #D1D4DC;
}

.panel-icon:hover {
  color: #1890ff;
}

.signature-form {
  display: block;
  padding: 1.5rem;
}

.signature-label {
  font-size: 0.6875rem;
  line-height: 100%;
  padding: 0.75rem;
  padding-left: 0;
}

.signature-preview {
  display: flex;
  height: 6.25rem;
  align-items: center;
  gap: 0.6875rem;
  align-self: stretch;
  border-radius: 6px;
  background: #D2D2D2;
  color: #D2D2D2;
  font-size: 3rem;
  line-height: 100%;
  justify-content: center;
}

.dancing-script, .mr-dafoe, .great-vibes {
  color: #000;
}

.dancing-script {
  font-family: 'Dancing Script';
}

.mr-dafoe {
  font-family: 'Mr Dafoe';
}

.great-vibes {
  font-family: 'Great Vibes';
}

.edit-user-info {
  padding: 1.5rem;
}

.edit-user-info-title {
  display: flex;
  flex-direction: column;
  justify-content: center;
}