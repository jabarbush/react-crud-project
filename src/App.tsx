import React, { useState } from "react";
import "./App.css";
import List from "./components/ListUsers";
import AddUser from "./components/AddUser";
import User from "./interfaces/User";
import userData from './db.json';

//I'm only *slighly* self-conscious that this file has too much logic in it
function App() {
  const [isPopupOpen, setIsPopupOpen] = useState(false);
  const [userList, setUserList] = useState<User[]>(userData.users);

  const handleOpenPopup = () => {
    setIsPopupOpen(true);
  };

  const handleClosePopup = () => {
    setIsPopupOpen(false);
  };

  const handleAddUser = (user: User) => {
    setUserList(prevList => [...prevList, user]);
    setIsPopupOpen(false);
  };

  return (
    <div className="app">
      <h1 className="page-title">User Roster</h1>
      <div className="toolbar">
        <button onClick={handleOpenPopup} className="add-user-btn">Add New User</button>
      </div>
      <List userList={userList} />
      {isPopupOpen &&
        <div className="popup-overlay">
          <AddUser onClose={handleClosePopup} onAddUser={handleAddUser} />
        </div>
      }
    </div>
  );
}

export default App;
