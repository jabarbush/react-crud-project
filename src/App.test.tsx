import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';

beforeEach(() => jest.resetAllMocks())
test('renders App component correctly', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders with no errors', () => {
  render(<App />);
});

test('renders the Toolbar component', () => {
  render(<App />);
  const toolbarComponent = screen.getByTestId('toolbar-component');
  expect(toolbarComponent).toBeInTheDocument();
});

test('renders the List component', () => {
  render(<App />);
  const listComponent = screen.getByTestId('list-component');
  expect(listComponent).toBeInTheDocument();
});
