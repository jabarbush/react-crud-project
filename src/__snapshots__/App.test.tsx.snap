// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      data-testid="toolbar-component"
    >
      <h1
        class="page-title"
      >
        User Roster
      </h1>
      <div
        class="toolbar"
      >
        <button
          class="add-user-btn"
        >
          Add New User
        </button>
        <div
          class="sort-container"
        >
          <h3
            class="toolbar-title"
          >
            Sort By: 
          </h3>
          <button
            class="sort-button"
          >
            Date of Birth (Ascending)
          </button>
          <h3
            class="toolbar-title"
          >
            Filter By: 
          </h3>
          <button
            class="sort-button"
          >
            Filter Options
          </button>
        </div>
      </div>
    </div>
    <div
      class="list-container "
    >
      <div
        data-testid="list-component"
      >
        <div
          class="list-container"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
